File: mailway_ui/lib/data/entity/QuoteMessage.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mailwayui/data/entity/PayloadKind.dart';

part 'QuoteMessage.g.dart';

@JsonSerializable()
class QuoteMessage {
  final String id;
  final int created_at;
  final int updated_at;
  final String chatMessageId;
  final String message_id;
  final String digest;
  final PayloadKind digest_kind;
  final String digest_description;
  final String sender_name;
  final String sender_public_key;
  final String chat_message_id;

  QuoteMessage({
    this.id,
    this.created_at,
    this.updated_at,
    this.chatMessageId,
    this.message_id,
    this.digest,
    this.digest_kind,
    this.digest_description,
    this.sender_name,
    this.sender_public_key,
    this.chat_message_id,
  });


  factory QuoteMessage.fromJson(Map<String, dynamic> json) =>
      _$QuoteMessageFromJson(json);

  Map<String, dynamic> toJson() => _$QuoteMessageToJson(this);
}

File: mailwaycore/src/main/java/com/dimension/mailwaycore/data/entity/QuoteMessage.kt
package com.dimension.mailwaycore.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.fasterxml.jackson.annotation.JsonIgnore
import kotlinx.serialization.Serializable

@Entity
@Serializable
data class QuoteMessage(
    @PrimaryKey
    @JsonIgnore
    var id: String,
    var created_at: Long,
    var updated_at: Long,
    var chatMessageId: String,
    var message_id: String,
    var digest: String,
    var digest_kind: PayloadKind,
    var digest_description: String,
    var sender_name: String,
    var sender_public_key: String
)